crates/router/src/workflows/outgoing_webhook_retry.rs

```diff
@@ -0,0 +1,374 @@
+    consumer::{self, workflows::ProcessTrackerWorkflow},
+    types::process_data,
+    utils as scheduler_utils,
+};
+
+use crate::{
+    core::webhooks::{self as webhooks_core, types::OutgoingWebhookTrackingData},
+    db::StorageInterface,
+    errors, logger,
+    routes::AppState,
+    types::{domain, storage},
+};
+
+pub struct OutgoingWebhookRetryWorkflow;
+
+#[async_trait::async_trait]
+impl ProcessTrackerWorkflow<AppState> for OutgoingWebhookRetryWorkflow {
+    #[instrument(skip_all)]
+    async fn execute_workflow<'a>(
+        &'a self,
+        state: &'a AppState,
+        process: storage::ProcessTracker,
+    ) -> Result<(), errors::ProcessTrackerError> {
+        let tracking_data: OutgoingWebhookTrackingData = process
+            .tracking_data
+            .clone()
+            .parse_value("OutgoingWebhookTrackingData")?;
+
+        let db = &*state.store;
+        let key_store = db
+            .get_merchant_key_store_by_merchant_id(
+                &tracking_data.merchant_id,
+                &db.get_master_key().to_vec().into(),
+            )
+            .await?;
+        let merchant_account = db
+            .find_merchant_account_by_merchant_id(&tracking_data.merchant_id, &key_store)
+            .await?;
+        let business_profile = db
+            .find_business_profile_by_profile_id(&tracking_data.business_profile_id)
+            .await?;
+
+        let event_id = format!(
+            "{}_{}",
+            tracking_data.primary_object_id, tracking_data.event_type
+        );
+        let event = db.find_event_by_event_id(&event_id).await?;
+
+        let (content, event_type) = get_outgoing_webhook_content_and_event_type(
+            state.clone(),
+            merchant_account.clone(),
+            key_store,
+            &tracking_data,
+        )
+        .await?;
+
+        match event_type {
+            // Resource status is same as the event type of the current event
+            Some(event_type) if event_type == tracking_data.event_type => {
+                let outgoing_webhook = OutgoingWebhook {
+                    merchant_id: tracking_data.merchant_id.clone(),
+                    event_id: event_id.clone(),
+                    event_type,
+                    content: content.clone(),
+                    timestamp: event.created_at,
+                };
+
+                webhooks_core::trigger_appropriate_webhook_and_raise_event(
+                    state.clone(),
+                    merchant_account,
+                    business_profile,
+                    outgoing_webhook,
+                    webhooks_core::types::WebhookDeliveryAttempt::AutomaticRetry,
+                    content,
+                    event_id,
+                    event_type,
+                    Some(process),
+                )
+                .await;
+            }
+            // Resource status has changed since the event was created, finish task
+            _ => {
+                logger::warn!(
```

Comment:
Agreed, we are skipping out on a few events when a resource transitions from a status for which webhooks are supported, to another status for which webhooks are not supported (by hyperswitch). As of today, such transitions are not _expected_ to happen, so a resource is always expected to transition from one status to another where webhooks for both statuses are supported. (But I believe nothing strictly prevents such a transition happening.)

However, to address this in a cleaner fashion, I'll be making modifications to the `events` table to allow storing the request and response payloads. That way, when retrying a webhook, we always fetch the request payload from the `events` table and send the request, instead of fetching the current status of the resource from the database. I'll be taking it up in a separate PR.